src/libYARP_sig/include/yarp/sig/IntrinsicParams.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_SIG_INTRINSICPARAMS_H
+#define YARP_SIG_INTRINSICPARAMS_H
+
+#include <yarp/os/Log.h>
+#include <yarp/os/Portable.h>
+#include <yarp/os/Property.h>
+#include <yarp/sig/api.h>
+
+namespace yarp {
+namespace sig {
+/**
+ * @brief The YarpDistortion enum to define the
+ * type of the distortion model of the camera.
+ *
+ * In geometric optics, distortion is a deviation from rectilinear projection;
+ * a projection in which straight lines in a scene remain straight in an image.
+ * Although distortion can be irregular or follow many patterns, the most commonly
+ * encountered distortions are radially symmetric. These can be represent by models
+ * that allow us to undistort the images knowing the parameters of the distortion
+ * (k1, k2, t1, t2, k3).
+ */
+enum class YarpDistortion : std::int32_t
```

Comment:
Why did you specify `std::int32_t`?
[C++ Core guideline](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#enum7-specify-the-underlying-type-of-an-enumeration-only-when-necessary) suggest to do it only when it is strictly necessary.
Are we enforcing always to be 32 bit?